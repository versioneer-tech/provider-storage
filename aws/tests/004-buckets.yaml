# Copyright 2025, EOX (https://eox.at) and Versioneer (https://versioneer.at)
# SPDX-License-Identifier: Apache-2.0

---
apiVersion: pkg.internal/v1beta1
kind: Storage
metadata:
  name: s-john
spec:
  bucketAccessGrants:
  - bucketName: s-john
    grantedAt: "2025-09-29T10:28:00Z"
    grantee: jane
    permission: None
  bucketAccessRequests:
  - bucketName: s-joe
    reason: Need access
    requestedAt: "2025-09-29T10:25:00Z"
  - bucketName: s-jeff
    reason: Need access
    requestedAt: "2025-09-29T10:26:00Z"
  - bucketName: s-jane
    reason: Need access
    requestedAt: "2025-09-29T10:27:00Z"
  buckets:
  - bucketName: s-john
    discoverable: true
  principal: john
status:
  conditions:
  - lastTransitionTime: "2024-01-01T00:00:00Z"
    message: 'Unready resources: access-key-john, bucket-s-john, observe-john-s-jane,
      and 6 more'
    reason: Creating
    status: "False"
    type: Ready
---
apiVersion: iam.aws.m.upbound.io/v1beta1
kind: AccessKey
metadata:
  annotations:
    crossplane.io/composition-resource-name: access-key-john
  labels:
    crossplane.io/composite: s-john
  name: john
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-john
    uid: ""
spec:
  forProvider:
    userRef:
      name: john
  providerConfigRef:
    kind: ProviderConfig
    name: provider-aws
  writeConnectionSecretToRef:
    name: john
---
apiVersion: s3.aws.m.upbound.io/v1beta1
kind: Bucket
metadata:
  annotations:
    crossplane.io/composition-resource-name: bucket-s-john
  labels:
    crossplane.io/composite: s-john
  name: s-john
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-john
    uid: ""
spec:
  forProvider:
    region: eu-central-1
  providerConfigRef:
    kind: ProviderConfig
    name: provider-aws
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: observe-john-s-jane
  labels:
    crossplane.io/composite: s-john
  name: observe-john-s-jane
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-john
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: iam.aws.m.upbound.io/v1beta1
      kind: Policy
      metadata:
        name: john.s-jane
        namespace: default
  managementPolicies:
  - Observe
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: observe-john-s-jeff
  labels:
    crossplane.io/composite: s-john
  name: observe-john-s-jeff
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-john
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: iam.aws.m.upbound.io/v1beta1
      kind: Policy
      metadata:
        name: john.s-jeff
        namespace: default
  managementPolicies:
  - Observe
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: observe-john-s-joe
  labels:
    crossplane.io/composite: s-john
  name: observe-john-s-joe
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-john
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: iam.aws.m.upbound.io/v1beta1
      kind: Policy
      metadata:
        name: john.s-joe
        namespace: default
  managementPolicies:
  - Observe
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: iam.aws.m.upbound.io/v1beta1
kind: Policy
metadata:
  annotations:
    crossplane.io/composition-resource-name: policy-jane-s-john
  labels:
    crossplane.io/composite: s-john
  name: jane.s-john
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-john
    uid: ""
spec:
  forProvider:
    policy: |-
      {
        "Version": "2012-10-17",
        "Statement": []
      }
  providerConfigRef:
    kind: ProviderConfig
    name: provider-aws
---
apiVersion: iam.aws.m.upbound.io/v1beta1
kind: Policy
metadata:
  annotations:
    crossplane.io/composition-resource-name: policy-s-john
  labels:
    crossplane.io/composite: s-john
  name: john.s-john
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-john
    uid: ""
spec:
  forProvider:
    policy: |-
      {
        "Version": "2012-10-17",
        "Statement": [
          {
            "Effect": "Allow",
            "Action": [
              "s3:*"
            ],
            "Resource": [
              "arn:aws:s3:::s-john",
              "arn:aws:s3:::s-john/*"
            ]
          }
        ]
      }
  providerConfigRef:
    kind: ProviderConfig
    name: provider-aws
---
apiVersion: iam.aws.m.upbound.io/v1beta1
kind: UserPolicyAttachment
metadata:
  annotations:
    crossplane.io/composition-resource-name: upa-s-john
  labels:
    crossplane.io/composite: s-john
  name: john.s-john
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-john
    uid: ""
spec:
  forProvider:
    policyArnRef:
      name: john.s-john
    userRef:
      name: john
  providerConfigRef:
    kind: ProviderConfig
    name: provider-aws
---
apiVersion: iam.aws.m.upbound.io/v1beta1
kind: User
metadata:
  annotations:
    crossplane.io/composition-resource-name: user-john
  labels:
    crossplane.io/composite: s-john
  name: john
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-john
    uid: ""
spec:
  forProvider: {}
  providerConfigRef:
    kind: ProviderConfig
    name: provider-aws
