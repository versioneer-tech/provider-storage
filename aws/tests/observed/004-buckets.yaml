---
apiVersion: s3.aws.upbound.io/v1beta2
kind: Bucket
metadata:
  annotations:
    crossplane.io/claim-name: alice-004
    crossplane.io/claim-namespace: default
    crossplane.io/composition-resource-name: bucket-alice-004-alice-004
spec: {}
status:
  conditions:
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced
---
apiVersion: s3.aws.upbound.io/v1beta2
kind: Bucket
metadata:
  annotations:
    crossplane.io/claim-name: alice-004
    crossplane.io/claim-namespace: default
    crossplane.io/composition-resource-name: bucket-alice-004-alice-shared-004
spec: {}
status:
  conditions:
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced
---
apiVersion: iam.aws.upbound.io/v1beta1
kind: Policy
metadata:
  annotations:
    crossplane.io/claim-name: alice-004
    crossplane.io/claim-namespace: default
    crossplane.io/composition-resource-name: policy-alice-004.owner.alice-004
spec: {}
status:
  conditions:
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced
---
apiVersion: iam.aws.upbound.io/v1beta1
kind: Policy
metadata:
  annotations:
    crossplane.io/claim-name: alice-004
    crossplane.io/claim-namespace: default
    crossplane.io/composition-resource-name: policy-alice-004.owner.alice-shared-004
spec: {}
status:
  conditions:
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced
---
apiVersion: kubernetes.crossplane.io/v1
kind: Object
metadata:
  annotations:
    crossplane.io/claim-name: alice-004
    crossplane.io/claim-namespace: default
    crossplane.io/composition-resource-name: check-policy-alice-004.readwrite.bob-shared-004
spec: {}
status:
  atProvider:
    manifest: {}
  conditions:
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced
---
apiVersion: iam.aws.upbound.io/v1beta1
kind: User
metadata:
  annotations:
    crossplane.io/claim-name: alice-004
    crossplane.io/claim-namespace: default
    crossplane.io/composition-resource-name: user-alice-004
spec: {}
status:
  conditions:
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced
---
apiVersion: iam.aws.upbound.io/v1beta1
kind: UserPolicyAttachment
metadata:
  annotations:
    crossplane.io/claim-name: alice-004
    crossplane.io/claim-namespace: default
    crossplane.io/composition-resource-name: user-policy-attachement-alice-004.owner.alice-004
spec: {}
status:
  conditions:
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced
---
apiVersion: iam.aws.upbound.io/v1beta1
kind: UserPolicyAttachment
metadata:
  annotations:
    crossplane.io/claim-name: alice-004
    crossplane.io/claim-namespace: default
    crossplane.io/composition-resource-name: user-policy-attachement-alice-004.owner.alice-shared-004
spec: {}
status:
  conditions:
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced
---
apiVersion: iam.aws.upbound.io/v1beta1
kind: UserPolicyAttachment
metadata:
  annotations:
    crossplane.io/claim-name: alice-004
    crossplane.io/claim-namespace: default
    crossplane.io/composition-resource-name: user-policy-attachement-alice-004.readwrite.bob-shared-004
spec: {}
status:
  conditions:
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced
---
apiVersion: iam.aws.upbound.io/v1beta1
kind: AccessKey
metadata:
  annotations:
    crossplane.io/claim-name: alice-004
    crossplane.io/claim-namespace: default
    crossplane.io/composition-resource-name: access-key-alice-004
spec: {}
status:
  conditions:
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced
