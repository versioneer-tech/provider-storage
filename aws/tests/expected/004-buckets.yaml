---
apiVersion: pkg.internal/v1beta1
kind: XStorage
metadata:
  labels:
    crossplane.io/claim-name: alice-004
    crossplane.io/claim-namespace: default
  name: alice-004
spec:
  bucketAccessRequests:
  - bucketName: bob-shared-004
    permission: ReadWrite
  buckets:
  - bucketName: alice-004
    discoverable: false
  - bucketName: alice-shared-004
    discoverable: true
  owner: alice-004
status:
  conditions:
  - lastTransitionTime: "2024-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
---
apiVersion: iam.aws.upbound.io/v1beta1
kind: AccessKey
metadata:
  annotations:
    crossplane.io/composition-resource-name: access-key-alice-004
  generateName: alice-004-
  labels:
    crossplane.io/composite: alice-004
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: XStorage
    name: alice-004
    uid: ""
spec:
  forProvider:
    userRef:
      name: alice-004
  providerConfigRef:
    name: storage-aws
  writeConnectionSecretToRef:
    name: alice-004
    namespace: default
---
apiVersion: s3.aws.upbound.io/v1beta2
kind: Bucket
metadata:
  annotations:
    crossplane.io/composition-resource-name: bucket-alice-004-alice-004
    xstorages.pkg.internal/discoverable: "false"
  generateName: alice-004-
  labels:
    crossplane.io/composite: alice-004
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: XStorage
    name: alice-004
    uid: ""
spec:
  forProvider:
    region: eu-central-1
  providerConfigRef:
    name: storage-aws
---
apiVersion: s3.aws.upbound.io/v1beta2
kind: Bucket
metadata:
  annotations:
    crossplane.io/composition-resource-name: bucket-alice-004-alice-shared-004
    xstorages.pkg.internal/discoverable: "true"
  generateName: alice-004-
  labels:
    crossplane.io/composite: alice-004
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: XStorage
    name: alice-004
    uid: ""
spec:
  forProvider:
    region: eu-central-1
  providerConfigRef:
    name: storage-aws
---
apiVersion: kubernetes.crossplane.io/v1alpha2
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: check-policy-alice-004.readwrite.bob-shared-004
  generateName: alice-004-
  labels:
    crossplane.io/composite: alice-004
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: XStorage
    name: alice-004
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: iam.aws.upbound.io/v1beta1
      kind: Policy
      metadata:
        name: alice-004.readwrite.bob-shared-004
  managementPolicies:
  - Observe
  providerConfigRef:
    name: storage-kubernetes
---
apiVersion: iam.aws.upbound.io/v1beta1
kind: Policy
metadata:
  annotations:
    crossplane.io/composition-resource-name: policy-alice-004.owner.alice-004
  generateName: alice-004-
  labels:
    crossplane.io/composite: alice-004
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: XStorage
    name: alice-004
    uid: ""
spec:
  forProvider:
    policy: |
      {
        "Version": "2012-10-17",
        "Statement": [
          {
            "Effect": "Allow",
            "Action": [ "s3:*" ],
            "Resource": [
              "arn:aws:s3:::<no value>",
              "arn:aws:s3:::<no value>/*"
            ]
          }
        ]
      }
  providerConfigRef:
    name: storage-aws
---
apiVersion: iam.aws.upbound.io/v1beta1
kind: Policy
metadata:
  annotations:
    crossplane.io/composition-resource-name: policy-alice-004.owner.alice-shared-004
  generateName: alice-004-
  labels:
    crossplane.io/composite: alice-004
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: XStorage
    name: alice-004
    uid: ""
spec:
  forProvider:
    policy: |
      {
        "Version": "2012-10-17",
        "Statement": [
          {
            "Effect": "Allow",
            "Action": [ "s3:*" ],
            "Resource": [
              "arn:aws:s3:::<no value>",
              "arn:aws:s3:::<no value>/*"
            ]
          }
        ]
      }
  providerConfigRef:
    name: storage-aws
---
apiVersion: iam.aws.upbound.io/v1beta1
kind: User
metadata:
  annotations:
    crossplane.io/composition-resource-name: user-alice-004
  generateName: alice-004-
  labels:
    crossplane.io/composite: alice-004
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: XStorage
    name: alice-004
    uid: ""
spec:
  forProvider: {}
  providerConfigRef:
    name: storage-aws
---
apiVersion: iam.aws.upbound.io/v1beta1
kind: UserPolicyAttachment
metadata:
  annotations:
    crossplane.io/composition-resource-name: user-policy-attachement-alice-004.owner.alice-004
  generateName: alice-004-
  labels:
    crossplane.io/composite: alice-004
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: XStorage
    name: alice-004
    uid: ""
spec:
  forProvider:
    policyArnRef:
      name: alice-004.owner.alice-004
    userRef:
      name: alice-004
  providerConfigRef:
    name: storage-aws
---
apiVersion: iam.aws.upbound.io/v1beta1
kind: UserPolicyAttachment
metadata:
  annotations:
    crossplane.io/composition-resource-name: user-policy-attachement-alice-004.owner.alice-shared-004
  generateName: alice-004-
  labels:
    crossplane.io/composite: alice-004
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: XStorage
    name: alice-004
    uid: ""
spec:
  forProvider:
    policyArnRef:
      name: alice-004.owner.alice-shared-004
    userRef:
      name: alice-004
  providerConfigRef:
    name: storage-aws
---
apiVersion: iam.aws.upbound.io/v1beta1
kind: UserPolicyAttachment
metadata:
  annotations:
    crossplane.io/composition-resource-name: user-policy-attachement-alice-004.readwrite.bob-shared-004
  generateName: alice-004-
  labels:
    crossplane.io/composite: alice-004
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: XStorage
    name: alice-004
    uid: ""
spec:
  forProvider:
    policyArnRef:
      name: alice-004.readwrite.bob-shared-004
    userRef:
      name: alice-004
  providerConfigRef:
    name: storage-aws
